packages/contracts/src/core/plugin/proposal/Proposal.sol

```diff
@@ -4,25 +4,72 @@ pragma solidity 0.8.17;
 
 import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
 
-import "./ProposalBase.sol";
+import "./IProposal.sol";
 
 /// @title Proposal
 /// @author Aragon Association - 2022-2023
-/// @notice An abstract contract defining the traits and internal functionality to create and execute proposals for non-upgradeable contracts.
-abstract contract Proposal is ProposalBase {
+/// @notice An abstract contract containing the traits and internal functionality to create and execute proposals that can be inherited by non-upgradeable DAO plugins.
+abstract contract Proposal is IProposal {
     using Counters for Counters.Counter;
 
     /// @notice The incremental ID for proposals and executions.
     Counters.Counter private proposalCounter;
 
-    /// @inheritdoc ProposalBase
+    /// @inheritdoc IProposal
     function proposalCount() public view override returns (uint256) {
         return proposalCounter.current();
     }
 
-    /// @inheritdoc ProposalBase
-    function createProposalId() internal override returns (uint256 proposalId) {
+    /// @notice Creates a proposal ID.
+    /// @return proposalId The proposal ID.
+    function _createProposalId() internal virtual returns (uint256 proposalId) {
```

Comment:
If we don't make it `virtual`, we're enforcing devs to always follow the `increment` id pattern. Let's say, we make it virtual. In that case, devs can follow `non-increment` id pattern. We could say that this might be out-of-pattern for `subgraph` as in subgraph we got a pattern how we connect proposal to `execute` of `dao`. But for our plugins, we're safe as we use normal `_createProposalId` (increment id pattern)

For others, if they make a new plugin, whats the plan ? should we still write subgraph for them ? i guess no which means they will write their own subgraph and use their own pattern.

Seems to me making this function `virtual` is a good idea. @brickpop  ?

Reply:
As discussed in our meeting, I removed `virtual`